crates/typst-library/src/visualize/image/mod.rs

```diff
@@ -224,13 +224,15 @@ impl Image {
         data: Bytes,
         format: ImageFormat,
         alt: Option<EcoString>,
-    ) -> StrResult<Image> {
+        span: Span,
```

Comment:
As discussed on Discord, we should avoid passing spans and file IDs into memoized functions if they are only used for formatting the error message. Rather, we'd have a return type that is not `SourceResult` and turn it into a `SourceResult` using span & ID outside. (Adding this to the PR discussion for completeness' sake.)